api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerInstanceTypePriceTypeAndCloudRegionTest.java

```diff
@@ -0,0 +1,707 @@
+    private static final float COMPUTE_PRICE_PER_HOUR = 11F;
+    private static final float DISK_PRICE_PER_HOUR = 1F;
+    private static final String INSTANCE_TYPE = "m5.large";
+    private static final String NOT_ALLOWED = "not allowed";
+    private static final String NO_PERMISSIONS = "no permissions";
+    private static final long PARENT_RUN_ID = 5L;
+    private static final String INSTANCE_DISK = "1";
+    private static final String PARENT_RUN_ID_PARAMETER = "parent-id";
+    private static final String PERMISSION_NAME = "READ";
+    private static final String SPOT = PriceType.SPOT.getLiteral();
+    private static final String ON_DEMAND = PriceType.ON_DEMAND.getLiteral();
+    private static final String DEFAULT_COMMAND = "sleep";
+    private static final String TEST_USER = "user";
+
+    @InjectMocks
+    private final PipelineRunManager pipelineRunManager = new PipelineRunManager();
+
+    @Mock
+    private ToolManager toolManager;
+
+    @Mock
+    private AuthManager securityManager;
+
+    @Mock
+    private RunPermissionManager permissionManager;
+
+    @Mock
+    private DataStorageManager dataStorageManager;
+
+    @Mock
+    private PipelineConfigurationManager pipelineConfigurationManager;
+
+    @Mock
+    private MessageHelper messageHelper;
+
+    @Mock
+    private InstanceOfferManager instanceOfferManager;
+
+    @Mock
+    private PipelineLauncher pipelineLauncher;
+
+    @Mock
+    private CloudRegionManager cloudRegionManager;
+
+    @Mock
+    private ToolVersionManager toolVersionManager;
+
+    @Mock
+    private CheckPermissionHelper permissionHelper;
+
+    @Mock
+    private PipelineRunDao pipelineRunDao;
+
+    @Mock
+    private PreferenceManager preferenceManager;
+
+    private static final String TEST_IMAGE = "testImage";
+    private Tool tool;
+    private PipelineConfiguration configuration;
+    private PipelineConfiguration configurationWithoutRegion;
+    private InstancePrice price;
+    private AwsRegion defaultAwsRegion;
+    private AwsRegion nonAllowedAwsRegion;
+    private AwsRegion notDefaultAwsRegion;
+    private PipelineRun parentRun;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        tool = getTool(TEST_IMAGE, DEFAULT_COMMAND);
+        defaultAwsRegion = getDefaultAwsRegion(ID);
+        nonAllowedAwsRegion = getNonDefaultAwsRegion(ID_2);
+        notDefaultAwsRegion = getNonDefaultAwsRegion(ID_3);
+        configuration = getPipelineConfiguration(TEST_IMAGE, INSTANCE_DISK, true, defaultAwsRegion.getId());
+        configurationWithoutRegion = getPipelineConfiguration(TEST_IMAGE, INSTANCE_DISK);
+        price = new InstancePrice(configuration.getInstanceType(), parseInt(configuration.getInstanceDisk()),
+                PRICE_PER_HOUR, COMPUTE_PRICE_PER_HOUR, DISK_PRICE_PER_HOUR);
+        parentRun = PipelineCreatorUtils.getPipelineRunWithInstance(PARENT_RUN_ID, TEST_USER, ID_3);
```

Comment:
Could you please join field declarations and assignments?

Reply:
I can't join them, as I use those variables among all the test class and test methods. So